android-components/components/feature/fxsuggest/src/test/java/mozilla/components/feature/fxsuggest/FxSuggestIngestionWorkerTest.kt

```diff
@@ -21,64 +22,63 @@ import org.junit.Assert.assertThrows
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.Mockito.doNothing
+import org.mockito.Mockito.doThrow
 import org.mockito.Mockito.verify
 import java.io.IOException
+import java.lang.RuntimeException
 
 @RunWith(AndroidJUnit4::class)
 class FxSuggestIngestionWorkerTest {
-    private lateinit var suggestStore: SuggestStore
+    private lateinit var storage: FxSuggestStorage
 
     @Before
     fun setUp() {
-        suggestStore = mock()
-        GlobalFxSuggestDependencyProvider.suggestStore = suggestStore
+        storage = mock()
+        GlobalFxSuggestDependencyProvider.storage = storage
     }
 
     @After
     fun tearDown() {
-        GlobalFxSuggestDependencyProvider.suggestStore = null
+        GlobalFxSuggestDependencyProvider.storage = null
     }
 
     @Test
     fun workSucceeds() = runTest {
-        doNothing().`when`(suggestStore).ingest(any())
+        whenever(storage.ingest(any())).thenReturn(Unit)
 
         val worker = TestListenableWorkerBuilder<FxSuggestIngestionWorker>(testContext).build()
 
         val result = worker.startWork().await()
 
-        verify(suggestStore).ingest(any())
+        verify(storage).ingest(any())
         assertEquals(ListenableWorker.Result.success(), result)
     }
 
     @Test
     fun workShouldRetry() = runTest {
-        whenever(suggestStore.ingest(any())).then {
+        whenever(storage.ingest(any())).then {
             throw SuggestApiException.Other("Mercury in retrograde")
         }
 
         val worker = TestListenableWorkerBuilder<FxSuggestIngestionWorker>(testContext).build()
 
         val result = worker.startWork().await()
 
-        verify(suggestStore).ingest(any())
+        verify(storage).ingest(any())
         assertEquals(ListenableWorker.Result.retry(), result)
     }
 
     @Test
-    fun workShouldRethrow() {
-        whenever(suggestStore.ingest(any())).then {
+    fun workShouldRethrow() = runTest {
```

Comment:
This test is failing now—`expected java.io.IOException to be thrown, but nothing was thrown`; the `IOException` either isn't thrown, or doesn't bubble up—and I can't figure out a way to make it pass! 😅

Reply:
I think this should do it:

```
        assertThrows(IOException::class.java) {
            runBlocking {
                worker.startWork().await()
            }
        }
```

Reply:
Omg, thank you! I completely forgot that `startWork` is a suspend function, but `assertThrows` isn't (in JUnit 4; it looks like JUnit 5 [supports suspend functions out of the box](https://github.com/junit-team/junit5/issues/1851)) 🎉